File: airbyte-config/config-models/src/main/java/io/airbyte/config/StreamResetRecord.java
/*
 * Copyright (c) 2020-2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.config;

import jakarta.annotation.Nullable;
import java.util.UUID;
import lombok.NonNull;

/**
 * A stream reset record is a reference to a stream that has a reset pending or running.
 **/
public record StreamResetRecord(@NonNull UUID connectionId,
                                @NonNull String streamName,
                                @Nullable String streamNamespace) {

}

File: airbyte-config/config-models/src/main/kotlin/io/airbyte/config/StreamResetRecord.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.config

import java.util.UUID

/**
 * A stream reset record is a reference to a stream that has a reset pending or running.
 */
data class StreamResetRecord(
  val connectionId: UUID,
  val streamName: String,
  val streamNamespace: String?,
)
